mod/wikirate_source/lib/javascript/script_new_source_page.js.coffee

```diff
@@ -1,14 +1,25 @@
-decko.slotReady (slot) ->
-  if slot.hasClass('new-view') && (slot.hasClass('TYPE-source') || slot.hasClass('TYPE-metric_value'))
-# bind listener to the tab anchor
-    slot.find('#myTab').find("a").click (e) ->
-# update the source type
-      source_type = $(this).data('source-type')
-      source_type_subcard = slot.find(".RIGHT-Xsource_type").find("input:hidden.d0-card-content")
-      source_type_subcard.val("[[" + source_type + "]]")
-      source_type_subcard.siblings("ul").find("input[value=" + source_type + "]").prop("checked", true)
-
 $(window).ready ->
+  $('body').on 'shown.bs.tab', 'form.TYPE-source.new-view .new-source-widget a[data-toggle="tab"]', (e) ->
+    source_type = $(e.target).data('source-type')
+    source_type_subcard = $(e.target).closest("form").find(".RIGHT-Xsource_type input:hidden.d0-card-content")
+
+    source_type_subcard.val("[[" + source_type + "]]")
+    source_type_subcard.siblings("ul").find("input[value=" + source_type + "]").prop("checked", true)
+
```

Comment:
as a general pattern, I'm thinking we might want to put these kinds of things into functions.  Mostly because that would make this easier to read.  

If we don't have that, we should probably at least put a comment, eg before the `on` trigger.